stdlib/public/TensorFlow/Ops.swift

```diff
@@ -674,7 +674,8 @@ public extension Tensor {
     wrt: self, vjp: _vjpTransposed(withPermutations:)
     where Scalar : TensorFlowFloatingPoint
   )
-  func transposed(withPermutations permutations: [Int32]) -> Tensor {
+  func transposed(withPermutations permutations: [Int]) -> Tensor {
```

Comment:
FWIW, the `Transpose` also supports `int64` as permutations. We just need to change the bindings at some point.

Reply:
Since `Raw.transpose` is generic, how about also making the `Tensor` permutations variant of `func transposed` be generic?
```swift
  func transposed<Index : BinaryInteger & TensorFlowScalar>(
    withPermutations permutations: Tensor<Int32>
  ) -> Tensor {
    return Raw.transpose(self, perm: permutations)
  }
```

Then, the `[Int]` permutations variant can map to `Int64`.

Reply:
Not sure I like that idea. If you did that, one may argue that you should do the same to all index-taking operators including subscript. This isn't consistent with the standard library APIs.

I meant the same as https://github.com/apple/swift/pull/24012#discussion_r275562253. The change I wanted to suggest is making the `map` on line 678 be `map(Int64.init)`.

Reply:
We discussed in person. We will keep `Int32` for now, since the `permutations: Tensor<Int32>` variant takes a `Tensor<Int32>`.